test/e2eexternal/dataplane/connectivity.go

```diff
@@ -0,0 +1,335 @@
+	containerName, ip string, port int) error {
+	// call `curl -m 10 {ip}:{port}` in container {containerName} of pod {podName} in namespace {namespace}
+	cmd := []string{"curl", "-m", "10", fmt.Sprintf("%s:%d", ip, port)}
+	stdout, stderr, err := runCommandInPod(cs, config, namespace, podName, containerName, cmd)
+	if err != nil {
+		return fmt.Errorf("failed to run command %v in %s %s %s: %v, stdout: %v, stderr: %v", cmd,
+			namespace, podName, containerName, err, stdout, stderr)
+	}
+
+	return nil
+}
+
+func runCommandInPod(cs *kubeclientset.Clientset, config *rest.Config, namespace, podName,
+	containerName string, cmd []string) (string, string, error) {
+	req := cs.CoreV1().RESTClient().Post().Resource("pods").Name(podName).Namespace(namespace).
+		SubResource("exec").Param("container", containerName)
+
+	req.VersionedParams(&v1.PodExecOptions{
+		Container: containerName,
+		Command:   cmd,
+		Stdin:     false,
+		Stdout:    true,
+		Stderr:    true,
+		TTY:       false,
+	}, scheme.ParameterCodec)
+
+	executor, err := remotecommand.NewSPDYExecutor(config, "POST", req.URL())
+	if err != nil {
+		return "", "", err
+	}
+
+	var stdout, stderr bytes.Buffer
+
+	err = executor.Stream(remotecommand.StreamOptions{
+		Stdin:  nil,
+		Stdout: &stdout,
+		Stderr: &stderr,
+		Tty:    false,
+	})
+	if err != nil {
+		return "", "", err
+	}
+
+	return stdout.String(), stderr.String(), nil
+}
+
+func getLogFromPod(cs *kubeclientset.Clientset, namespace, podName, containerName string) (string, error) {
+	req := cs.CoreV1().RESTClient().Get().Namespace(namespace).Name(podName).Resource("pods").SubResource("log")
+	req.Param("follow", strconv.FormatBool(false)).
+		Param("container", containerName).
+		Param("previous", strconv.FormatBool(false)).
+		Param("timestamps", strconv.FormatBool(false))
+
+	closer, err := req.Stream()
+	if err != nil {
+		return "", err
+	}
+
+	defer closer.Close()
+
+	out := new(strings.Builder)
+
+	_, err = io.Copy(out, closer)
+	if err != nil {
+		return "", err
+	}
+
+	return out.String(), nil
+}
+
+func getDockerIP(name, networkName string) (string, error) {
+	var stdout bytes.Buffer
+
+	cmdargs := []string{"inspect", name, "-f",
+		fmt.Sprintf("{{(index .NetworkSettings.Networks \"%s\").IPAddress}}", networkName),
+	}
+	cmd := exec.Command("docker", cmdargs...)
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
+		return "", fmt.Errorf("failed to run command docker %v: %v", cmdargs, err)
+	}
+
+	// output has trailing "\n", so it needs to be trimed
+	return strings.TrimSuffix(stdout.String(), "\n"), nil
+}
+
+func getLogFromDocker(name string) (string, error) {
+	var stderr bytes.Buffer
+	// get stderr of `docker log {name}` command
+	cmd := exec.Command("docker", "logs", name)
+	cmd.Stderr = &stderr
+	if err := cmd.Run(); err != nil {
+		return "", fmt.Errorf("failed to run docker logs %s: %v", name, err)
+	}
+
+	return stderr.String(), nil
+}
```

Comment:
docker.go ? :)

Reply:
I think this no longer exists, has an ancestor in `scripts/e2e/external/utils`.